src/NHibernate.Test/Async/NHSpecificTest/NH3583/AutoFlushFixture.cs

```diff
@@ -61,9 +61,12 @@ protected override void OnTearDown()
 				Assert.That(result.Count, Is.EqualTo(1));
 			}
 		}
+
 		[Test]
 		public async Task ShouldAutoFlushWhenInDistributedTransactionAsync()
 		{
+			Assume.That(Sfi.ConnectionProvider.Driver.SupportsSystemTransactions);
```

Comment:
There is a discrepancy here with the sync version, which gets changed on async regen.

Reply:
Fixed